### PR TITLE
(proof-new) Add proofs for sine secant lemmas.

### PR DESCRIPTION
**Pull request by nafur**
This PR adds proofs for secant lemmas for the sine function.

### CODE REVIEW

**Comment by nafur** [comment]
```
Results of [regress]: 0 / 2323 failed
Results of [eager_checking]: 2 / 2323 failed
Results of [check_proofs]: 2 / 2323 failed
Results of [check_proofs_p1]: 307 / 2323 failed {TRUST, THEORY_LEMMA}
Results of [check_proofs_p2]: 328 / 2323 failed {ARRAYS_TRUST, INT_TRUST, STRING_TRUST}
Results of [check_proofs_p10]: 482 / 2323 failed {(THEORY_)PREPROCESS, (THEORY_)PREPROCESS_LEMMA, WITNESS_AXIOM}
Results of [check_proofs_fail]: 1012 / 2323 failed
```

File: src/expr/proof_rule.h [hunk 1]
```diff
@@ -1226,6 +1226,36 @@ enum class PfRule : uint32_t
...
+  // t, calculated as follows:
+  //    (p(l) - p(u)) / (l - u) * (t - l) + p(l)
+  // The lemma states that if t is between l and u, then (sin t) is below the
+  // secant of p from l to u.
+  ARITH_TRANS_SINE_APPROX_ABOVE_NEG,
+  //======== Sine is approximated from below for positive values
+  // Children: none
+  // Arguments: (d, t, lb, ub, l, u)
+  // ---------------------
+  // Conclusion: (=> (and (>= t lb) (<= t ub)) (>= (sine t) (secant sin l u t))
```
Comment (ajreynol) [inline src/expr/proof_rule.h:1248]:
```suggestion
  // Conclusion: (=> (and (>= t lb) (<= t ub)) (>= (sin t) (secant sin l u t))
```

File: src/expr/proof_rule.h [hunk 2]
```diff
@@ -1226,6 +1226,36 @@ enum class PfRule : uint32_t
   //   (=> (< t pi) (< (sin t) (- pi t)))
   // )
   ARITH_TRANS_SINE_TANGENT_PI,
+  //======== Sine is approximated from above for negative values
+  // Children: none
+  // Arguments: (d, t, lb, ub, l, u)
+  // ---------------------
+  // Conclusion: (=> (and (>= t lb) (<= t ub)) (<= (sine t) (secant sin l u t))
```
Comment (ajreynol) [inline src/expr/proof_rule.h:1233]:
```suggestion
  // Conclusion: (=> (and (>= t lb) (<= t ub)) (<= (sin t) (secant sin l u t))
```

**Review by ajreynol — APPROVED** [review]
Minor comments.

**Comment by nafur** [comment]
After the merge:
```
Results of [regress]: 0 / 2322 failed
Results of [eager_checking]: 2 / 2322 failed
Results of [check_proofs]: 2 / 2322 failed
Results of [check_proofs_p1]: 306 / 2322 failed {TRUST, THEORY_LEMMA}
Results of [check_proofs_p2]: 327 / 2322 failed {ARRAYS_TRUST, INT_TRUST, STRING_TRUST}
Results of [check_proofs_p10]: 481 / 2322 failed {(THEORY_)PREPROCESS, (THEORY_)PREPROCESS_LEMMA, WITNESS_AXIOM}
Results of [check_proofs_fail]: 1012 / 2322 failed
```

**Pull request merged**